core/client/src/main/java/alluxio/client/file/FileInStream.java

```diff
@@ -89,10 +95,10 @@
   private final boolean mShouldCache;
 
   // The following 3 fields must be kept in sync. They are only updated in updateStreams together.
-  /** Current {@link BlockInStream} backing this stream. */
-  protected BlockInStream mCurrentBlockInStream;
-  /** Current {@link BufferedBlockOutStream} writing the data into Alluxio. */
-  protected BufferedBlockOutStream mCurrentCacheStream;
+  /** Current block in stream backing this stream. */
+  protected InputStream mCurrentBlockInStream;
+  /** Current block out stream writing the data into Alluxio. */
+  protected OutputStream mCurrentCacheStream;
```

Comment:
Should these types be more advanced so we can avoid casting them throughout the class?